scripts/setup.js

```diff
@@ -0,0 +1,178 @@
+
+  updatePackageJSON(packageJSON, tsdxName, pluginName)
+  updateJBrowseConfig(tsdxName, pluginName)
+  updateExampleFixture(tsdxName, pluginName)
+  makeJBrowseDir()
+
+  if (!alreadyRun) {
+    setupGithubAction(packageJSON, projectName)
+  }
+}
+
+function updatePackageJSON(packageJSON, tsdxName, pluginName) {
+  // 1. Change "name" in the "jbrowse-plugin" field to the name of your project (e.g. "MyProject")
+  packageJSON['jbrowse-plugin'].name = pluginName
+
+  // 2. In the "scripts" field, replace the default name with the name of your project, prefixed with "JBrowsePlugin" in the "start" and "build" entries
+  packageJSON.scripts.start = `tsdx watch --verbose --noClean --format umd --name JBrowsePlugin${pluginName} --onFirstSuccess "yarn serve --cors --listen 9000 ."`
+
+  packageJSON.scripts.build = `tsdx build --format cjs,esm,umd --name JBrowsePlugin${pluginName}`
+
+  // 3. In the "module" field, replace jbrowse-plugin-my-project with the name of your project (leave off the @myscope if using a scoped package name) (you can double-check that the filename is correct after running the build step below and comparing the filename to the file in the dist/ folder)
+  packageJSON.module = `dist/${tsdxName}.esm.js`
+
+  // this overwrites package.json
+  writeJSON(packageJSON, 'package.json')
+}
+
+// replace default plugin name and url with project name and dist file
+function updateJBrowseConfig(tsdxName, pluginName) {
+  const jbrowseConfig = require('../jbrowse_config.json')
+  jbrowseConfig.plugins[0].name = pluginName
+  jbrowseConfig.plugins[0].url = `http://localhost:9000/dist/${tsdxName}.umd.development.js`
+  writeJSON(jbrowseConfig, 'jbrowse_config.json')
+}
+
+// replace default plugin name and url with project name and dist file
+function updateExampleFixture(tsdxName, pluginName) {
+  const exampleFixture = require('../cypress/fixtures/hello_view.json')
+  exampleFixture.plugins[0].name = pluginName
+  exampleFixture.plugins[0].url = `http://localhost:9000/dist/${tsdxName}.umd.development.js`
+  writeJSON(exampleFixture, 'cypress/fixtures/hello_view.json')
+}
+
+// create a dot directory for the jbrowse build to live in
+function makeJBrowseDir() {
+  if (!fs.existsSync('.jbrowse')) {
+    fs.mkdirSync('.jbrowse')
+  }
+}
+
+function setupGithubAction(packageJSON, projectName) {
+  // move integration test into workflow folder
+  if (!fs.existsSync(path.join('.github', 'workflows'))) {
+    fs.mkdirSync(path.join('.github', 'workflows'), { recursive: true })
+  }
+  fs.renameSync(
+    'integration.yml',
+    path.join('.github', 'workflows', 'integration.yml'),
+  )
+
+  // add status badge to README
+  const repoUrl = parseRepoUrl(packageJSON.repository)
+  if (repoUrl !== undefined) {
+    let README = readFile('README.md').split(/\r?\n/)
+    README[0] = `# ${projectName} ![Integration](${repoUrl}/workflows/Integration/badge.svg?branch=main)`
+    fs.writeFileSync('README.md', README.join(os.EOL), 'utf8')
+  }
+}
+
+/* 
```

Comment:
Trailing whitespace here

```suggestion
/*
```